app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,50 @@
+class BookmarksController < ApplicationController
+  def new
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = @topic.bookmarks.new
+  end
+
+  def create
+     @topic = Topic.find(params[:topic_id])
+     @bookmark = @topic.bookmarks.build(bookmark_params)
+     current_user.bookmarks(topic_id: @topic) << @bookmark
```

Comment:
Does this work?

I would expect this to fail because the `bookmarks` association doesn't accept arguments in this way.

Perhaps you were thinking of adding a bookmark to the current_user association. So something like this...

```ruby
current_user.bookmarks << @bookmark
```